### ISSUE TITLE
Write up shortcomings of CODEOWNERS parser

### ISSUE THREAD

**Issue by cbart** [sourcegraph/sourcegraph#46898]
We've talked about that there are a few known things that our codeowners parser today doesn't support. While I don't think we need to support _everything_ today, it would be great to have an exhaustive list of things that we need to do eventually, probably as a new issue on this board.

**Comment by cbart** [comment]
Started writing down the semantics of the parser and matcher: https://docs.google.com/document/d/1Ev-pDqIF75hyesI6qp9DHbbo83E7c9gnRhTW6M5MIhc/edit#

**Comment by cbart** [comment]
Some current shortcomings:

- The parser does not work with Bitbucket examples (GitHub and GitLab works fine) - specifically inline team resolution
- The evaluator (when you call `FindOwners`) will construct all the regexps and other pattern-matching-objects every time `FindOwners` is called after loading the file. So the file is parsed (into a Protocol Buffer) but the `FindOwners` calls could save some time by making matchers only once.
